package.json

```diff
@@ -40,13 +41,14 @@
         "@comet/eslint-plugin": "workspace:*",
         "@formatjs/cli": "^6.0.0",
         "@types/node": "^22.0.0",
+        "knip": "^5.38.1",
         "cspell": "^8.15.4",
         "dotenv-cli": "^7.0.0",
         "husky": "^7.0.4",
         "lint-staged": "^12.3.7",
         "prettier": "^2.0.0",
         "rimraf": "^3.0.2",
-        "typescript": "^4.2.2",
+        "typescript": "^5.7.2",
```

Comment:
Why do you need to upgrade TS? I believe this was tried before and caused some problems.

Reply:
that's needed because knip has a minimum peer dependency of 5.0.4

<img width="367" alt="image" src="https://github.com/user-attachments/assets/cf8e5ace-e019-4302-a173-b124630abe71">